runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -123,6 +142,58 @@ MM_SchedulingDelegate::getInitialTaxationThreshold(MM_EnvironmentVLHGC *env)
 	return getNextTaxationThreshold(env);
 }
 
+void 
+MM_SchedulingDelegate::globalMarkCycleStart(MM_EnvironmentVLHGC *env){
+	calculateGlobalMarkOverhead(env);
+
+	_currentlyPerformingGMP = true;
+	/* Reset the total time taken for each increment of global mark phase, along with the time for concurrent mark GC work*/
+	_globalMarkIncrementsTotalTime = 0;
+	_concurrentMarkGCThreadsTotalWorkTime = 0;
+}
+
+void 
+MM_SchedulingDelegate::calculateGlobalMarkOverhead(MM_EnvironmentVLHGC *env) {
+	/* Calculate statistics regarding GMP overhead */
+	PORT_ACCESS_FROM_ENVIRONMENT(env);
+
+	/* Determine how long it has been since previous global mark cycle started */
+	uint64_t globalMarkIntervalEndTime = j9time_hires_clock();
+	uint64_t globalMarkIntervalTime = j9time_hires_delta(_globalMarkIntervalStartTime, globalMarkIntervalEndTime, J9PORT_TIME_DELTA_IN_MICROSECONDS);
```

Comment:
shouldn't this compare both end points (or both start points)?

Reply:
It's comparing both start points. At the end of this very function, we have the following line

```
/* Set start time of next GMP phase, as end of current one */
_globalMarkIntervalStartTime = globalMarkIntervalEndTime;
```